src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -297,24 +302,28 @@ public Task Stop()
         /// </summary>
         public void Abort()
         {
-            _requestProcessingStopping = true;
-            _requestAborted = true;
+            TransitionToState(FrameState.Aborted);
 
             _requestBody?.Abort();
             _responseBody?.Abort();
 
             try
             {
-                ConnectionControl.End(ProduceEndType.SocketDisconnect);
-                SocketInput.AbortAwaiting();
-                RequestAbortedSource.Cancel();
+                ConnectionControl?.End(ProduceEndType.SocketDisconnect);
```

Comment:
Is `ConnectionControl` ever set to null? Same question for the other additions of `?.`